src/Microsoft.ComponentDetection.Orchestrator/Experiments/ExperimentService.cs

```diff
@@ -1,44 +1,63 @@
-﻿namespace Microsoft.ComponentDetection.Orchestrator.Experiments;
+namespace Microsoft.ComponentDetection.Orchestrator.Experiments;
 
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Common.DependencyGraph;
 using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.ComponentDetection.Orchestrator.ArgumentSets;
 using Microsoft.ComponentDetection.Orchestrator.Experiments.Configs;
 using Microsoft.ComponentDetection.Orchestrator.Experiments.Models;
+using Microsoft.ComponentDetection.Orchestrator.Services;
+using Microsoft.ComponentDetection.Orchestrator.Services.GraphTranslation;
 using Microsoft.Extensions.Logging;
 
 /// <inheritdoc />
 public class ExperimentService : IExperimentService
 {
     private readonly ConcurrentDictionary<IExperimentConfiguration, ExperimentResults> experiments;
     private readonly IEnumerable<IExperimentProcessor> experimentProcessors;
+    private readonly IGraphTranslationService graphTranslationService;
     private readonly ILogger<ExperimentService> logger;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="ExperimentService"/> class.
     /// </summary>
     /// <param name="configs">The experiment configurations.</param>
     /// <param name="experimentProcessors">The experiment processors.</param>
+    /// <param name="graphTranslationService">The graph translation service.</param>
     /// <param name="logger">The logger.</param>
     public ExperimentService(
         IEnumerable<IExperimentConfiguration> configs,
         IEnumerable<IExperimentProcessor> experimentProcessors,
+        IGraphTranslationService graphTranslationService,
         ILogger<ExperimentService> logger)
     {
         this.experiments = new ConcurrentDictionary<IExperimentConfiguration, ExperimentResults>(
             configs.ToDictionary(config => config, _ => new ExperimentResults()));
         this.experimentProcessors = experimentProcessors;
+        this.graphTranslationService = graphTranslationService;
         this.logger = logger;
     }
 
     /// <inheritdoc />
-    public void RecordDetectorRun(IComponentDetector detector, IEnumerable<DetectedComponent> components)
+    public void RecordDetectorRun(IComponentDetector detector, ComponentRecorder componentRecorder, IDetectionArguments detectionArguments)
```

Comment:
nit: use interfaces in method signatures
```suggestion
    public void RecordDetectorRun(IComponentDetector detector, IComponentRecorder componentRecorder, IDetectionArguments detectionArguments)
```

Reply:
We cannot pass the interface, because `DetectorProcessingResult` requires `ComponentRecorder`, not `IComponentRecorder`

https://github.com/microsoft/component-detection/blob/262fa8857d16fa133264678db17d8b1f5229617c/src/Microsoft.ComponentDetection.Orchestrator/Services/DetectorProcessingResult.cs#L13